### PR TITLE
Fixed NaN path issue on rounded corners for animated arcs (arcTween)

### PR DESCRIPTION
**Pull request by wassx**
d3_geom_polygonIntersect ran into a division by 0. A check for undefined result allows to continue with a defined value. Please check if i have not run in a misunderstanding of the function.

related issue: https://github.com/mbostock/d3/issues/2183

### CODE REVIEW

File: src/geom/polygon.js [hunk 1]
```diff
@@ -94,6 +94,7 @@ function d3_geom_polygonIntersect(c, d, a, b) {
   var x1 = c[0], x3 = a[0], x21 = d[0] - x1, x43 = b[0] - x3,
```
Comment (gardenamacloc900) [inline src/geom/polygon.js]: Yes

**Comment by mgold** [comment]
`ua = ua || 1` is a few characters shorter.

**Comment by Tiagoperes** [comment]
Will this be merged into master?

**Comment by mbostock** [comment]
A different fix is merged into #2591.

**Pull request closed without merging**